recipes/lora_finetune_distributed.py

```diff
@@ -139,12 +134,10 @@ def __init__(self, cfg: DictConfig) -> None:
         self.total_epochs = cfg.epochs
         self.max_steps_per_epoch = cfg.max_steps_per_epoch
         self.global_step = 0
-        self._resume_from_checkpoint = cfg.resume_from_checkpoint
+
         self._save_adapter_weights_only = cfg.get("save_adapter_weights_only", False)
+        self._resume_from_checkpoint = cfg.resume_from_checkpoint
         self._gradient_accumulation_steps = cfg.gradient_accumulation_steps
-        self._fsdp_sharding_strategy = torch.distributed.fsdp.ShardingStrategy[
```

Comment:
See [this comment](https://github.com/pytorch/torchtune/pull/1287#discussion_r1711980968)

Reply:
Thanks. This is annoying since now if you don't want sharding you have to use a completely different set of modules. My only remaining comment then is that people know that FSDP has 3 levels (also coming from DeepSpeed) and I don't think it's clear that Full shard and opt shard are controlled by reshard_after_forward. Could we make the config name different?

Reply:
want to call out gradient and opt state are shaded regardless of reshard_after_forward=true/false. reshard_after_forward mainly wants to avoid all-gather in the backward